client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/WatcherStatsResponseTests.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.client.NodesResponseHeader;
+import org.elasticsearch.client.NodesResponseHeaderTestUtils;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.test.ESTestCase;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+
+public class WatcherStatsResponseTests extends ESTestCase {
+
+    public void testFromXContent() throws IOException {
+        xContentTester(
+            this::createParser,
+            this::createTestInstance,
+            this::toXContent,
+            WatcherStatsResponse::fromXContent)
+            .supportsUnknownFields(true)
+            .randomFieldsExcludeFilter(field -> field.endsWith("stats"))
+            .test();
+    }
+
+    private void toXContent(WatcherStatsResponse response, XContentBuilder builder) throws IOException {
+        builder.startObject();
+        NodesResponseHeaderTestUtils.toXContent(response.getHeader(), response.getClusterName(), builder);
```

Comment:
This is only used here, do we need a separate class for it?

Reply:
I expect tests for other NodesAction-based rest actions will be using this as well in the future. We could have used it [here](https://github.com/elastic/elasticsearch/blob/a086c665a35af32e60e6d23ab6ca75fcf085b46f/client/rest-high-level/src/test/java/org/elasticsearch/client/security/ClearRolesCacheResponseTests.java#L41) for example if we had it handy.